`Development`: Document the manual assessment of programming exercises (#8968)
ls1intum/Artemis: docs/user/exercises/programming.rst

This also means that other config files, like SwiftLint's ``.swiftlint.yml`` file, are not shown. Those files currently
can only be accessed via your own IDE.

Manual Assessment
-----------------

There are two ways to manually assess programming exercises: with the Orion plugin, you can assess the student's code directly in your JetBrains IDE. Refer to the :ref:`Orion plugin documentation <orion>` for more information.
Alternatively, you can use the online editor to assess the students' submissions.

.. figure:: programming/online-editor-assessment.png
   :alt: Manual assessment in the online editor
   :align: center

   Manual assessment in the online editor

For manual assessment, the online editor provides the following features, tailored to the assessment process:

> 1. **File browser**: The file browser shows the files in the student's submission and allows you to navigate between them. Changed files are highlighted.

Reviewer: ```suggestion
1. **File browser**: The file browser shows the files in the student's submission and allows you to navigate between them. Changed files are highlighted in yellow.
```

Author: Added

> 2. **Build output**: The build output shows the output of the build process, if any. This is useful in case of build errors.

Reviewer: ```suggestion
2. **Build output**: The build output shows the output of the build process, if any exists. This is useful in case of build errors.
```

3. **Read-only editor**: The read-only editor allows you to view the student's code. It highlights all changed lines.

> 4. **Instructions**: The instructions provide the problem statement, including the tasks the student has solved successfully as determined by the test cases of the exercise, if any. Furthermore, any structured assessment criteria are listed here.

Reviewer: I'd also mention the test case feedback in the top right as well as the linked test cases in the problem statement

Author: I'll add something

> You can add feedback directly in the source code by first hovering over the line on which you want to comment and then clicking on the |add-feedback-button| button that appears next to the line numbers. You can also add general feedback using the (todo add button) button at the bottom of the page.

Reviewer: ```suggestion
You can add feedback directly in the source code by first hovering over the line on which you want to comment and then clicking on the |add-feedback-button| button that appears next to the line number. You can also add general feedback using the ACTUALLY ADD THE BUTTON button at the bottom of the page.
```
I think you missed a todo here

Reviewer: Maybe also make it clearer that the student will also see the inline feedback in that place

Author: Good catch. I'll move the sentence and add the file

.. figure:: programming/online-editor-feedback.png
   :alt: Feedback in the online editor
   :align: center

   Editing feedback in the online editor

After clicking the button, you can enter a feedback comment and a score for the selected line. Alternatively, you can drag one of the structured assessment criteria from the instructions to the feedback area to apply it to the feedback widget.
When you are done, click the |save-feedback-button| button to save your feedback or the |cancel-feedback-button| button to discard it.

.. figure:: programming/online-editor-general-feedback.png
   :alt: General feedback in the online editor
   :align: center

   General feedback in the online editor

> You can also add general feedback by going scrolling to the bottom of the page. Feedback given this way is not tied to any specific file or line of code.

Author: ```suggestion
You can also add general feedback by scrolling to the bottom of the page. Feedback given this way is not tied to any specific file or line of code.
```

This also means that other config files, like SwiftLint's ``.swiftlint.yml`` file, are not shown. Those files currently
can only be accessed via your own IDE.

Manual Assessment
-----------------

There are two ways to manually assess programming exercises: with the Orion plugin, you can assess the student's code directly in your JetBrains IDE. Refer to the :ref:`Orion plugin documentation <orion>` for more information.
Alternatively, you can use the online editor within Artemis to assess the students' submissions directly in the browser.

.. figure:: programming/online-editor-assessment.png
   :alt: Manual assessment in the online editor
   :align: center

   Manual assessment in the online editor

> For manual assessment, the online editor provides the following features, tailored to the assessment process:

Reviewer 2: ```suggestion
For manual assessment, the online editor provides the following features tailored to the assessment process:
```

1. **File browser**: The file browser shows the files in the student's submission and allows you to navigate between them. Changed files are highlighted in yellow.
2. **Build output**: The build output shows the output of the build process, if any. This is useful in case of build errors.
3. **Read-only editor**: The read-only editor allows you to view the student's code. It highlights all changed lines.
4. **Instructions**: The instructions provide the structured grading criteria and the problem statement, including the tasks the student has solved successfully as determined by the test cases of the exercise. You can review the test cases associated with each task by clicking on the number of passing tests next to the task, e.g. |problem-statement-task|
5. **Result**: The top right corner holds the current result of the student's submission. By clicking on it, you can review the test cases and feedback currently attached to the submission.

> You can add feedback directly in the source code by first hovering over the line you want to comment on and then clicking the |add-feedback-button| that appears next to the line number. Students will be able to see feedback added this way in the same location when viewing their assessed submission.

Author: ```suggestion
You can add feedback directly in the source code by first hovering over the line you want to comment on and then clicking the |add-feedback-button| that appears next to the line number. Students will be able to see feedback added this way in the same location when viewing their assessed submission in the online editor.
```
